### DISCUSSION TITLE
Wireable not updating on wire:model

### DISCUSSION THREAD

**Discussion by hendrikwjy** [livewire/livewire#3780] [7. Bugs (archived)]
1. I made DTO 

``` php
class Dto implements Wireable
{
    public $items = [];

    public function __construct($items)
    {
        $this->items = $items;
    }

    public function toLivewire()
    {
        return $this->items;
    }

    public static function fromLivewire($value)
    {
        ray($value);
        return new static($value);
    }
}
```
2. I initialize the Dto
``` php
 public function mount()
    {
        $this->data = new Dto([
            'kode' => 'Auto',
            'nama' => 'f',
            'aktif' => 1,
            'keterangan' => '',
        ]);

    }
```

3. I link it to wire:model
``` html
<input type="text" class="form-control " maxlength=50 placeholder="Nama" {{ $show==1 ? "readonly" : "" }} wire:model="data.nama" />

```

4. When i type in the input field the DTO is not updating, instead it create another field outside the DTO.
![image](https://user-images.githubusercontent.com/31981819/133709603-ed1bb9a0-70f3-42cb-bc38-1ba270cfb552.png)

**Comment by mabdullahsari** [comment]
If you look closely, it is not adding it "outside the DTO", but assigning it as a class property on your object.

You can probably fix this by implementing the `__set` magic method that internally sets the `items` array; making it a surrogate for your write operations on your object.

**Reply by francoism90** [reply]
@mabdullahsari I know this is old.. but I would like to point out you can also add `__get`, `__isset` as well. :)

**Comment by joshhanley** [comment]
@hendrikwjy you need to the properties of the DTO directly, so in your case it would be `wire:model="data.items.nama"`.

If you want it to just be `data.nama`, then you either need to add a property `$nama` to your DTO, or do something like @mabdullahsari suggested above.

Hope this helps!
